sonar-plugin-api/src/main/java/org/sonar/api/server/rule/DefaultRuleDescriptionSection.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule;
+
+import java.io.IOException;
+import java.net.URL;
+import org.apache.commons.io.IOUtils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class DefaultRuleDescriptionSection implements RuleDescriptionSection {
+
+  private final String key;
+  private final String htmlDescription;
+
+  private DefaultRuleDescriptionSection(String key, String htmlDescription) {
+    this.key = key;
+    this.htmlDescription = htmlDescription;
+  }
+
+  @Override
+  public String getKey() {
+    return key;
+  }
+
+  @Override
+  public String getHtmlDescription() {
+    return htmlDescription;
+  }
+
+  public static DefaultRuleDescriptionSectionBuilder builder() {
+    return new DefaultRuleDescriptionSectionBuilder();
+  }
+
+  public static final class DefaultRuleDescriptionSectionBuilder {
+    private String sectionKey;
+    private String htmlDescription;
+
+    private DefaultRuleDescriptionSectionBuilder() {
+    }
+
+    /**
+     * Identifier of the section, must be unique for a given rule
```

Comment:
```suggestion
     * Identifier of the section, must be unique across sections of a given rule
```
If I understood your intent correctly.
"unique for a given rule" could be interpreted as "no two rules should have the same section identifier", and I guess you meant "no two sections in a rule should have the same identifier"

Reply:
yes indeed, thanks for the suggestion that's much clearer 👍